src/JetMETAnalyzer.cc

```diff
@@ -0,0 +1,517 @@
+  std::vector<float>& refy = tree["refy"].write<std::vector<float>>();
+  std::vector<float>& refarea = tree["refarea"].write<std::vector<float>>();
+  std::vector<float>& jte = tree["jte"].write<std::vector<float>>();
+  std::vector<float>& jtpt = tree["jtpt"].write<std::vector<float>>();
+  std::vector<float>& jteta = tree["jteta"].write<std::vector<float>>();
+  std::vector<float>& jtphi = tree["jtphi"].write<std::vector<float>>();
+  std::vector<float>& jtm = tree["jtm"].write<std::vector<float>>();
+  std::vector<float>& jty = tree["jty"].write<std::vector<float>>();
+  std::vector<float>& jtarea = tree["jtarea"].write<std::vector<float>>();
+  std::vector<float>& jtjec = tree["jtjec"].write<std::vector<float>>();
+
+  std::vector<float>& beta = tree["beta"].write<std::vector<float>>();
+  std::vector<float>& betaStar = tree["betaStar"].write<std::vector<float>>();
+  std::vector<float>& betaClassic = tree["betaClassic"].write<std::vector<float>>();
+  std::vector<float>& betaStarClassic = tree["betaStarClassic"].write<std::vector<float>>();
+  std::vector<float>& dZ = tree["dZ"].write<std::vector<float>>();
+  std::vector<float>& DRweighted = tree["DRweighted"].write<std::vector<float>>();
+  std::vector<float>& fRing0 = tree["fRing0"].write<std::vector<float>>();
+  std::vector<float>& fRing1 = tree["fRing1"].write<std::vector<float>>();
+  std::vector<float>& fRing2 = tree["fRing2"].write<std::vector<float>>();
+  std::vector<float>& fRing3 = tree["fRing3"].write<std::vector<float>>();
+  std::vector<float>& fRing4 = tree["fRing4"].write<std::vector<float>>();
+  std::vector<float>& fRing5 = tree["fRing5"].write<std::vector<float>>();
+  std::vector<float>& fRing6 = tree["fRing6"].write<std::vector<float>>();
+  std::vector<float>& fRing7 = tree["fRing7"].write<std::vector<float>>();
+  std::vector<float>& fRing8 = tree["fRing8"].write<std::vector<float>>();
+  std::vector<float>& nCh = tree["nCh"].write<std::vector<float>>();
+  std::vector<float>& nNeutrals = tree["nNeutrals"].write<std::vector<float>>();
+  std::vector<float>& ptD = tree["ptD"].write<std::vector<float>>();
+  std::vector<bool>& isMatched = tree["isMatched"].write<std::vector<bool>>();
+
+  nref = 0;
+  refdrjt.clear();
+  refrank.clear();
+  refpdgid_algorithmicDef.clear();
+  refpdgid_physicsDef.clear();
+  refpdgid.clear();
+  refe.clear();
+  refpt.clear();
+  refeta.clear();
+  refphi.clear();
+  refm.clear();
+  refy.clear();
+  refarea.clear();
+  jte.clear();
+  jtpt.clear();
+  jteta.clear();
+  jtphi.clear();
+  jtm.clear();
+  jty.clear();
+  jtarea.clear();
+  jtjec.clear();
+
+  beta.clear();
+  betaStar.clear();
+  betaClassic.clear();
+  betaStarClassic.clear();
+  dZ.clear();
+  DRweighted.clear();
+  fRing0.clear();
+  fRing1.clear();
+  fRing2.clear();
+  fRing3.clear();
+  fRing4.clear();
+  fRing5.clear();
+  fRing6.clear();
+  fRing7.clear();
+  fRing8.clear();
+  nCh.clear();
+  nNeutrals.clear();
+  ptD.clear();
+  isMatched.clear();
```

Comment:
If all of these clears happen each event it might be nice to move these to a beginEvent() function (or any other function that is not the analyze function). Although, that depends on either moving the references to the class declaration or a new object, as suggested above. Thoughts?